### ISSUE TITLE
receiving gamemode switch; no explosion

### ISSUE THREAD

**Issue by m-symons** [BigPapi13/Delta#3]
heya! first time using a datapack from github here. running on multiplayer lan vanilla 1.18.1 on hard mode, although this also happens on latest (1.19.2?) after updating the map this is running on.
running these functions:
tick_ramp (called every tick):
```
execute as @a[x=-5,y=-3,z=-93,dx=2,dy=2,dz=1] run function domos:trigger_ramp
```

trigger_ramp:
```
# base launch power
scoreboard players set $x delta.api.launch 0
scoreboard players set $y delta.api.launch 750
scoreboard players set $z delta.api.launch -7500
function delta:api/launch_xyz

# facing-based launch power to allow player to adjust aim
scoreboard players set $strength delta.api.launch 2500
function delta:api/launch_looking
```

when trigger_ramp is called by stepping into the target selector volume, no explosions seem to go off. the player isn't launched, and even without the resource pack on, there's no visible/audible explosions. the player is also set to creative, regardless of whether or not they were in creative to begin with. feels like one of the functions is giving up part-way through, but there's no errors in the console related to the pack at all.

might be a namespacing issue on my end, since i haven't been paying much attention to it so far. would appreciate any help though!
thanks!

**Comment by Calverin** [comment]
Hey @ralcore, the issue is likely from your command, try `execute as @a at @s if entity @s[x=-5,y=-3,z=-93,dx=2,dy=2,dz=1] run function domos:trigger_ramp`

**Comment by m-symons** [comment]
yeah, managed to get in touch with @BigPapi13 through discord. the lack of explosions was due to the location (thanks!) but the creative mode switching is a bug with the current implementation when calling more than one explosion on an entity per tick.
